indico/modules/receipts/util.py

```diff
@@ -294,3 +316,17 @@ def get_preview_placeholders(value):
         x[0].startswith('registration.field_data['),
     ))
     return placeholders
+
+
+def get_event_attachment_images(event: Event) -> dict[str, Attachment]:
+    """Get a dict of the event's attached images indexed by the attachment's ID."""
+    attached_items = get_attached_items(event)
+    all_attachments = attached_items.get('files', []).copy()
+    for folder in attached_items.get('folders', []):
+        all_attachments.extend(folder.attachments)
+    return {attachment.id: attachment for attachment in all_attachments if attachment.file.is_image}
+
+
+def is_image_supported(content_type: str) -> bool:
+    """Check if the given content type is an image supported by Pillow."""
+    return content_type.split('/')[1].upper() in Image.OPEN
```

Comment:
I think your env is broken 😄 

```
In [1]: from PIL import Image

In [2]: Image.OPEN
Out[2]: 
{'BLP': (PIL.BlpImagePlugin.BlpImageFile,
  <function PIL.BlpImagePlugin._accept(prefix)>),
 'BMP': (PIL.BmpImagePlugin.BmpImageFile,
  <function PIL.BmpImagePlugin._accept(prefix)>),
 'DIB': (PIL.BmpImagePlugin.DibImageFile,
  <function PIL.BmpImagePlugin._dib_accept(prefix)>),
 'BUFR': (PIL.BufrStubImagePlugin.BufrStubImageFile,
  <function PIL.BufrStubImagePlugin._accept(prefix)>),
 'CUR': (PIL.CurImagePlugin.CurImageFile,
  <function PIL.CurImagePlugin._accept(prefix)>),
 'PCX': (PIL.PcxImagePlugin.PcxImageFile,
  <function PIL.PcxImagePlugin._accept(prefix)>),
 'DCX': (PIL.DcxImagePlugin.DcxImageFile,
  <function PIL.DcxImagePlugin._accept(prefix)>),
 'DDS': (PIL.DdsImagePlugin.DdsImageFile,
  <function PIL.DdsImagePlugin._accept(prefix)>),
 'EPS': (PIL.EpsImagePlugin.EpsImageFile,
  <function PIL.EpsImagePlugin._accept(prefix)>),
 'FITS': (PIL.FitsImagePlugin.FitsImageFile,
  <function PIL.FitsImagePlugin._accept(prefix)>),
 'FLI': (PIL.FliImagePlugin.FliImageFile,
  <function PIL.FliImagePlugin._accept(prefix)>),
 'FTEX': (PIL.FtexImagePlugin.FtexImageFile,
  <function PIL.FtexImagePlugin._accept(prefix)>),
 'GBR': (PIL.GbrImagePlugin.GbrImageFile,
  <function PIL.GbrImagePlugin._accept(prefix)>),
 'GIF': (PIL.GifImagePlugin.GifImageFile,
  <function PIL.GifImagePlugin._accept(prefix)>),
 'GRIB': (PIL.GribStubImagePlugin.GribStubImageFile,
  <function PIL.GribStubImagePlugin._accept(prefix)>),
 'HDF5': (PIL.Hdf5StubImagePlugin.HDF5StubImageFile,
  <function PIL.Hdf5StubImagePlugin._accept(prefix)>),
 'PNG': (PIL.PngImagePlugin.PngImageFile,
  <function PIL.PngImagePlugin._accept(prefix)>),
 'JPEG2000': (PIL.Jpeg2KImagePlugin.Jpeg2KImageFile,
  <function PIL.Jpeg2KImagePlugin._accept(prefix)>),
 'ICNS': (PIL.IcnsImagePlugin.IcnsImageFile,
  <function PIL.IcnsImagePlugin._accept(prefix)>),
 'ICO': (PIL.IcoImagePlugin.IcoImageFile,
  <function PIL.IcoImagePlugin._accept(prefix)>),
 'IM': (PIL.ImImagePlugin.ImImageFile, None),
 'IMT': (PIL.ImtImagePlugin.ImtImageFile, None),
 'IPTC': (PIL.IptcImagePlugin.IptcImageFile, None),
 'JPEG': (<function PIL.JpegImagePlugin.jpeg_factory(fp=None, filename=None)>,
  <function PIL.JpegImagePlugin._accept(prefix)>),
 'MCIDAS': (PIL.McIdasImagePlugin.McIdasImageFile,
  <function PIL.McIdasImagePlugin._accept(s)>),
 'MPEG': (PIL.MpegImagePlugin.MpegImageFile, None),
 'TIFF': (PIL.TiffImagePlugin.TiffImageFile,
  <function PIL.TiffImagePlugin._accept(prefix)>),
 'MSP': (PIL.MspImagePlugin.MspImageFile,
  <function PIL.MspImagePlugin._accept(prefix)>),
 'PCD': (PIL.PcdImagePlugin.PcdImageFile, None),
 'PIXAR': (PIL.PixarImagePlugin.PixarImageFile,
  <function PIL.PixarImagePlugin._accept(prefix)>),
 'PPM': (PIL.PpmImagePlugin.PpmImageFile,
  <function PIL.PpmImagePlugin._accept(prefix)>),
 'PSD': (PIL.PsdImagePlugin.PsdImageFile,
  <function PIL.PsdImagePlugin._accept(prefix)>),
 'QOI': (PIL.QoiImagePlugin.QoiImageFile,
  <function PIL.QoiImagePlugin._accept(prefix)>),
 'SGI': (PIL.SgiImagePlugin.SgiImageFile,
  <function PIL.SgiImagePlugin._accept(prefix)>),
 'SPIDER': (PIL.SpiderImagePlugin.SpiderImageFile, None),
 'SUN': (PIL.SunImagePlugin.SunImageFile,
  <function PIL.SunImagePlugin._accept(prefix)>),
 'TGA': (PIL.TgaImagePlugin.TgaImageFile, None),
 'WEBP': (PIL.WebPImagePlugin.WebPImageFile,
  <function PIL.WebPImagePlugin._accept(prefix)>),
 'WMF': (PIL.WmfImagePlugin.WmfStubImageFile,
  <function PIL.WmfImagePlugin._accept(prefix)>),
 'XBM': (PIL.XbmImagePlugin.XbmImageFile,
  <function PIL.XbmImagePlugin._accept(prefix)>),
 'XPM': (PIL.XpmImagePlugin.XpmImageFile,
  <function PIL.XpmImagePlugin._accept(prefix)>),
 'XVTHUMB': (PIL.XVThumbImagePlugin.XVThumbImageFile,
  <function PIL.XVThumbImagePlugin._accept(prefix)>)}
  ```

Reply:
they also provide this btw, but you're supposed to always double check against `Image.OPEN` because some might not actually have a rendering plugin:
```
In [3]: Image.registered_extensions()
Out[3]: 
{'.blp': 'BLP',
 '.bmp': 'BMP',
 '.dib': 'DIB',
 '.bufr': 'BUFR',
 '.cur': 'CUR',
 '.pcx': 'PCX',
 '.dcx': 'DCX',
 '.dds': 'DDS',
 '.ps': 'EPS',
 '.eps': 'EPS',
 '.fit': 'FITS',
 '.fits': 'FITS',
 '.fli': 'FLI',
 '.flc': 'FLI',
 '.ftc': 'FTEX',
 '.ftu': 'FTEX',
 '.gbr': 'GBR',
 '.gif': 'GIF',
 '.grib': 'GRIB',
 '.h5': 'HDF5',
 '.hdf': 'HDF5',
 '.png': 'PNG',
 '.apng': 'PNG',
 '.jp2': 'JPEG2000',
 '.j2k': 'JPEG2000',
 '.jpc': 'JPEG2000',
 '.jpf': 'JPEG2000',
 '.jpx': 'JPEG2000',
 '.j2c': 'JPEG2000',
 '.icns': 'ICNS',
 '.ico': 'ICO',
 '.im': 'IM',
 '.iim': 'IPTC',
 '.jfif': 'JPEG',
 '.jpe': 'JPEG',
 '.jpg': 'JPEG',
 '.jpeg': 'JPEG',
 '.mpg': 'MPEG',
 '.mpeg': 'MPEG',
 '.tif': 'TIFF',
 '.tiff': 'TIFF',
 '.mpo': 'MPO',
 '.msp': 'MSP',
 '.palm': 'PALM',
 '.pcd': 'PCD',
 '.pdf': 'PDF',
 '.pxr': 'PIXAR',
 '.pbm': 'PPM',
 '.pgm': 'PPM',
 '.ppm': 'PPM',
 '.pnm': 'PPM',
 '.psd': 'PSD',
 '.qoi': 'QOI',
 '.bw': 'SGI',
 '.rgb': 'SGI',
 '.rgba': 'SGI',
 '.sgi': 'SGI',
 '.ras': 'SUN',
 '.tga': 'TGA',
 '.icb': 'TGA',
 '.vda': 'TGA',
 '.vst': 'TGA',
 '.webp': 'WEBP',
 '.wmf': 'WMF',
 '.emf': 'WMF',
 '.xbm': 'XBM',
 '.xpm': 'XPM'}
 ```

Reply:
This seems to be lazy-loaded. It's only being populated after I actually opened an image using `Image.open()`...

I'd rather just use our usual logic of checking if it's an image type we want to support that we also use in places where one can upload images:

```
            try:
                with file.open() as f:
                    pic = Image.open(f)
            except OSError:
                raise ValidationError(_('Invalid picture file.'))
            if pic.format.lower() not in {'jpeg', 'png', 'gif', 'webp'}:
                raise ValidationError(_('This field only accepts jpg, png, gif and webp picture formats.'))
```

MIME type information is not particularly trustworthy anyway, nor is the file extension.